### DISCUSSION TITLE
How to get all compression formats?

### DISCUSSION THREAD

**Discussion by bronger** [go-chi/chi#753] [Q&A] [answered]
I use github.com/go-chi/chi/v5/middleware.Compress(5).

I need to expose the supported compression encoders in the XML response of my server. (To be concrete, it is the `<compression>` tag in [OAI-PMH](http://www.openarchives.org/OAI/openarchivesprotocol.html#Identify).) How do I get them? I can hardcode ["gzip", "deflate"] but this might change in the future.

**Comment by VojtechVitek** [comment, marked as answer]
I think you can hardcode gzip and deflate. They will not go away.

The compress middleware doesn't have `GetEncoders()` method right now, perhaps it could be added. But then you'd have to instantiate your Compressor by hand.
